src/proposal.mligo

```diff
@@ -391,22 +396,23 @@ let flush(n, config, store : nat * config * storage): return =
 
 // Removes an accepted and finished proposal by key.
 let drop_proposal (proposal_key, config, store : proposal_key * config * storage): return =
-  let store = authorize_admin store in
-
   let proposal = check_if_proposal_exist (proposal_key, store) in
-  if is_voting_period_over(proposal, config.voting_period, store)
+  let proposal_is_expired = is_time_reached (proposal, config.proposal_expired_time) in
+
+  if   (sender = proposal.proposer)
+    || (sender = store.guardian && sender <> source) // Guardian cannot be equal to SOURCE
+    || proposal_is_expired
   then
-    if   do_total_vote_meet_quorum_threshold(proposal, store, config.quorum_threshold)
-      && proposal.upvotes > proposal.downvotes
-    then
-      let store = unfreeze_proposer_and_voter_token
-            (config.rejected_proposal_return_value, true, proposal, config.voting_period, config.fixed_proposal_fee_in_token, store) in
-      let store = delete_proposal (proposal.start_date, proposal_key, store) in
-      (([] : operation list), store)
-    else
-      (failwith("FAIL_DROP_PROPOSAL_NOT_ACCEPTED") : return)
+    let store = unfreeze_proposer_and_voter_token
+          ( config.rejected_proposal_return_value
+          , false // Expired proposal is treated as rejected proposal regardless of its actual votes
```

Comment:
I think it would be more clear to say:
```suggestion
          , false // A dropped proposal is treated as rejected regardless of its actual votes
```
Just because the same happens to proposal that are dropped before they are expired.

Reply:
Updated